File: parser/services/parser.js
import * as handler from "./handle.js";
import {load} from "cheerio";


const cities = ["astana","almaty","taldykorgan", "kostanai"]

export const GetExchangers = async () => {
    const data = []
    cities.map(async (item,key) =>  {
        const res = await GetExchangersByCity(item)
        data.push(res)
    })
    return data
}

export const GetExchangersByCity = async (city) => {
    const parsed_data = await handler.pageContent(`https://kurs.kz/index.php?mode=${city}`)
    if (!parsed_data) {
        return
    }
    const response = []
    for (let punkt of parsed_data) {
        let exchanger = {
            id: 0,
            name: punkt.name,
            city: punkt.city,
            address: punkt.address,
            wholesale: punkt.phone,
            updated_time: punkt.date,
            phone_numbers: punkt.phones,
            USD: punkt.data.USD,
            EUR: punkt.data.EUR,
            RUB: punkt.data.RUB,
        }
        response.push(exchanger)
    }
    return response
}

File: handler/pkg/models/parser.go
package models

type ParserResponse struct {
	Id           int        `json:"id"`
	Name         string     `json:"name"`
	City         string     `json:"city"`
	Address      string     `json:"address"`
	WholeSale    string     `json:"wholesale"`
	UpdatedTime  uint64     `json:"updated_time"`
	PhoneNumbers []string   `json:"phone_numbers"`
	USD          [2]float32 `json:"USD"`
	RUB          [2]float32 `json:"RUB"`
	EUR          [2]float32 `json:"EUR"`
}

type ExchangerCurrenciesResponse struct {
	ExchangerId int        `json:"-" db:"exchanger_id"`
	UploadTime  uint64     `json:"upload_time" db:"upload_time"`
	USD         [2]float32 `json:"USD" db:"usd"`
	EUR         [2]float32 `json:"EUR" db:"eur"`
	RUB         [2]float32 `json:"RUB" db:"rub"`
}
